libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -181,9 +185,9 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
      * @param routeDrawData a [Expected<RouteLineError, RouteSetValue>]
      */
     fun renderRouteDrawData(style: Style, routeDrawData: Expected<RouteLineError, RouteSetValue>) {
-        rebuildSourcesAndLayersIfNeeded(style)
-        jobControl.scope.launch(Dispatchers.Main) {
+        jobControl.scope.launch(Dispatchers.Main.immediate) {
```

Comment:
Why do you need immediate dispatcher here?

Reply:
I want to ensure that layers are initialized as soon as the `renderRouteDrawData` function (or any other function) returns, so that developers can synchronously reference these layers right away. Since `MapboxRouteLineView` is required to be interacted with on the main thread, ensuring that we're not posting the task back to the message queue is enough.

This is an optimization we should do either way really, knowing that some implementations can struggle with giving us access of the main thread back.

Reply:
But layers initialization is inside a lock. There's no guarantee that it will be synchronous. We can wait inside `launch` block for quite some time to acquire the lock.

Reply:
So we don't need a lock?

Reply:
I don't think so. At least I can't see it being required with the current codebase.